src/Http/Middleware/AddIdpUserDataMiddleware.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Zanichelli\IdpExtension\Http\Middleware;
+
+use Closure;
+use Exception;
+use GuzzleHttp\Client;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Zanichelli\IdpExtension\Models\ZTrait\ZUserBuilder;
+
+class AddIdpUserDataMiddleware
+{
+    use ZUserBuilder;
+
+    public function handle(Request $request, Closure $next, string $idpUrlEnvKey = 'IDP_URL')
+    {
+        $token = $request->cookies->get(config("idp.cookie.name"));
+        if ($token) {
+            try {
+                $client = new Client(['verify' => false]);
+                $res = $client->get(env($idpUrlEnvKey) . '/v1/user?token=' . $token);
+                $request->merge(['user' => json_decode($res->getBody(), true)]);
+            } catch (Exception $e) {
+                Log::error($e->getMessage());
+                return response()->json(['message' => $e->getMessage()], $e->getCode());
+            }
+        }
```

Comment:
Non dovrebbe restituire un 401 (o comunque un errore) in assenza del token?

Reply:
Però credo che eventualmente il codice per quell'errore venga restituito direttamente da IDP e noi qui poi lo restituiamo.

Reply:
Perché idp? Non viene neppure chiamato se il token non è presente.